src/client/dfs/dfs.c

```diff
@@ -1022,7 +1033,8 @@ open_sb(daos_handle_t coh, bool create, dfs_attr_t *attr, daos_handle_t *oh)
 		D_GOTO(err, rc = EINVAL);
 	}
 
-	if (iods[1].iod_size != sizeof(sb_ver) || sb_ver != DFS_SB_VERSION) {
+	/** check version compatibility */
+	if (iods[1].iod_size != sizeof(sb_ver) || sb_ver > DFS_SB_VERSION) {
```

Comment:
Minor/Prior Style (This wasn't introduced in this PR)
At first look, since `sb_ver` is set to 0 above, it seems that `sb_ver` will always be 0 in this check.
But further above, I see we're doing `d_iov_set(&sg_iovs[1], &sb_ver, sizeof(dfs_sb_ver_t));` and then after setting `sb_ver = 0` we make a call to `daos_obj_fetch`.
This seems a little confusing, though in hindsight I do understand what is going on.
Do we care about this? I.e. would it be better to initialize to 0 before
`d_iov_set(&sg_iovs[1], &sb_ver, sizeof(dfs_sb_ver_t));` to avoid confusion?

Reply:
I would think not, unless maybe for some erroneous valgrind issue.